lib/kubernetes-deploy/runner_task.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+require 'tempfile'
+
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/ui_helpers'
+require 'kubernetes-deploy/kubectl'
+
+module KubernetesDeploy
+  class RunnerTask
+    include KubeclientBuilder
+    include Kubectl
+    include UIHelpers
+
+    class FatalTaskRunError < FatalDeploymentError; end
+    class TaskTemplateMissingError < FatalDeploymentError
+      def initialize(task_template, namespace, context)
+        super("Pod template `#{task_template}` cannot be found in namespace: `#{namespace}`, context: `#{context}`")
+      end
+    end
+
+    def initialize(namespace:, context:, logger: KubernetesDeploy.logger)
+      @logger = logger
+      @namespace = namespace
+      @kubeclient = build_v1_kubeclient(context)
+      @context = context
+    end
+
+    def run(task_template:, entrypoint:, args:, env_vars: [])
+      phase_heading("Validating configuration")
+      validate_configuration(task_template, entrypoint, args)
+
+      phase_heading("Fetching task template")
+      raw_template = get_template(task_template)
+
+      phase_heading("Constructing final pod specification")
+      rendered_template = build_pod_template(raw_template, entrypoint, args, env_vars)
+
+      validate_pod_spec(rendered_template)
+
+      phase_heading("Creating pod")
+      KubernetesDeploy.logger.info("Starting task runner pod: '#{rendered_template.metadata.name}'")
+      @kubeclient.create_pod(rendered_template)
+    end
+
+    private
+
+    def validate_configuration(task_template, args)
+      errors = []
+
+      if task_template.blank?
+        errors << "Task template name can't be nil"
+      end
+
+      if @namespace.blank?
+        errors << "Namespace can't be empty"
+      end
+
+      if args.blank?
+        errors << "Args can't be nil"
+      end
+
+      namespace = begin
+        @kubeclient.get_namespace(namespace)
+      rescue KubeException
```

Comment:
`KubeException` could also mean that the auth didn't work or network is down. Let's check it the `error_code == 404` in the same manner as you do for other resources.